File: Annexe8-Exercice/app/src/main/java/com/example/annexe8_exercice/MainActivity.java
package com.example.annexe8_exercice;

import android.animation.ObjectAnimator;
import android.os.Bundle;
import android.view.View;
import android.widget.LinearLayout;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    ObjectAnimator objectAnimator;
    LinearLayout linearLayout;
    boolean ouvert = false;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        linearLayout = findViewById(R.id.linear);

        objectAnimator = ObjectAnimator.ofFloat(linearLayout, View.TRANSLATION_Y, -100);



        linearLayout.setOnClickListener(source-> {

            if(!ouvert){
                objectAnimator.start();
                ouvert = true;
            } else {
                objectAnimator.reverse();
                ouvert = false;
            }
        });
    }
}

File: Annexe3B-Serialisation/app/src/main/java/com/example/annexe3b_serialisation/MainActivity.kt
package com.example.annexe3b_serialisation

import android.os.Bundle
import android.view.View
import android.widget.SeekBar
import android.widget.SeekBar.OnSeekBarChangeListener
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    lateinit var sonnerie: SeekBar
    lateinit var media: SeekBar
    lateinit var notif: SeekBar

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        this.enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v: View, insets: WindowInsetsCompat ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        val singleton = Singleton.getInstance(applicationContext)

        sonnerie = findViewById(R.id.sonnerie)
        media = findViewById(R.id.media)
        notif = findViewById(R.id.notif)

        sonnerie.setProgress(singleton.getSonnerieProgress())
        media.setProgress(singleton.getMediaProgress())
        notif.setProgress(singleton.getNotifProgress())

        sonnerie.setOnSeekBarChangeListener(object : OnSeekBarChangeListener {
            override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
            }

            override fun onStartTrackingTouch(seekBar: SeekBar) {}

            override fun onStopTrackingTouch(seekBar: SeekBar) {
                singleton.setSeekBar( sonnerie.getProgress(),
                    media.getProgress(), notif.getProgress() )

                singleton.ajouterSon("sonnerie", sonnerie.getProgress())
                singleton.ajouterSon("media", media.getProgress())
                singleton.ajouterSon("notif", notif.getProgress())
            }
        })

        try {
            singleton.deserialiserListe()
            sonnerie.progress = singleton.getSonnerieProgress()
            media.progress = singleton.getMediaProgress()
            notif.progress = singleton.getNotifProgress()
        } catch (e: Exception) {
            e.printStackTrace();
        }

    }

    override fun onStop() {
        super.onStop()
        try {
            Singleton.getInstance(applicationContext).serialiserListe()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}